src/Baseline.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BackwardCompatibility;
+
+use function array_values;
+use function preg_match;
+
+/** @psalm-immutable */
+final class Baseline
+{
+    /** @psalm-param list<string> $ignoredChanges */
+    private function __construct(private readonly array $ignoredChanges = [])
+    {
+    }
+
+    public static function empty(): self
+    {
+        return new self();
+    }
+
+    public static function fromList(string ...$ignoredChanges): self
+    {
+        return new self(array_values($ignoredChanges));
+    }
+
+    public function ignores(Change $change): bool
+    {
+        foreach ($this->ignoredChanges as $ignoredChangeRegex) {
+            if (preg_match($ignoredChangeRegex, $change->__toString()) === 1) {
```

Comment:
Not that it would do tons of difference but we just need to run `__toString()` once.

Reply:
Updated :+1: